.github/workflows/tests-vmtests-osmodifier.yml

```diff
@@ -0,0 +1,199 @@
+            libvirt-daemon-kvm libvirt-devel qemu-kvm qemu-img python3-libvirt \
+            python3-devel edk2-ovmf \
+            azure-cli
+
+          sudo tdnf list installed
+
+          sudo systemctl restart libvirtd
+          sudo systemctl status libvirtd
+
+      - name: Install prerequisites (Ubuntu 24.04)
+        if: inputs.hostDistro == 'ubuntu2404'
+        run: |
+          set -eux
+
+          sudo apt update -y
+          sudo apt -y install python3-venv python3-pip python3-dev \
+              libvirt-dev libvirt-daemon libvirt-daemon-system libvirt-clients \
+              qemu-kvm virt-manager
+
+          # Install arm64 specific
+          if [[ "$HOST_ARCH" == "arm64" ]]; then
+            sudo apt -y install qemu-system-arm qemu-efi-aarch64 ovmf seabios
+          fi
+
+          sudo apt list --installed
+
+          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+        env:
+          HOST_ARCH: ${{ inputs.hostArch }}
+ 
+      - name: Azure Login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ vars.AZURE_CLIENT_ID }}
+          tenant-id: ${{ vars.AZURE_TENANT_ID }}
+          subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+        
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: repo
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: out
+
+      - name: Import container
+        id: importContainer
+        run: |
+          set -eux
+
+          CONTAINER_TAR_PATH="out/container-${{ inputs.hostArch }}/imagecustomizer.tar.gz"
+          DOCKER_OUTPUT=$(sudo docker image load -i "$CONTAINER_TAR_PATH" 2>&1)
+          CONTAINER_TAG=$(echo $DOCKER_OUTPUT | awk '{print $3}')
+
+          echo "containerTag=$CONTAINER_TAG" >> "$GITHUB_OUTPUT"
+        env:
+          HOST_ARCH: ${{ inputs.hostArch }}
+
+      - name: Verify osmodifier binary
+        id: osmodifier_bin
+        run: |
+          set -eux
+
+          OSMODIFIER_BIN="$(realpath out/osmodifier-${{ inputs.hostArch }}/osmodifier)"
+          [[ -f "$OSMODIFIER_BIN" ]] || { echo "osmodifier binary not found at $OSMODIFIER_BIN"; ls -la out/; exit 1; }
+          chmod +x "$OSMODIFIER_BIN"
+          echo "path=$OSMODIFIER_BIN" >> "$GITHUB_OUTPUT"
+
+      - name: Download base images
+        run: |
+          set -eux
+
+          ./repo/.github/workflows/scripts/download-image.sh "$AZURE_STORAGE" "$AZURE_CONTAINER" "azure-linux/core-efi-vhdx-2.0-$HOST_ARCH" azl-core-efi-2.0
+          ./repo/.github/workflows/scripts/download-image.sh "$AZURE_STORAGE" "$AZURE_CONTAINER" "azure-linux/core-efi-vhdx-3.0-$HOST_ARCH" azl-core-efi-3.0
+
+          if [ "$HOST_ARCH" == "amd64" ]; then
+            ./repo/.github/workflows/scripts/download-image.sh "$AZURE_STORAGE" "$AZURE_CONTAINER" "azure-linux/core-legacy-vhd-2.0-$HOST_ARCH" azl-core-legacy-2.0
```

Comment:
You probably don't need to download the legacy boot images (`image.vhd`). You can just test against the UEFI boot images (`image.vhdx`).